core/src/main/java/com/redhat/lightblue/client/LightblueClient.java

```diff
@@ -17,6 +17,6 @@
 
     LightblueBulkDataResponse bulkData(AbstractDataBulkRequest<AbstractLightblueDataRequest> requests) throws LightblueException;
 
-	<T> T data(AbstractLightblueDataRequest lightblueRequest, Class<T> type) throws LightblueException;
+	<T> T data(LightblueRequest lightblueRequest, Class<T> type) throws LightblueException;
```

Comment:
AbstractLightblueDataRequest is the parent of all the data request objects implemented in the lightblue client. Specifically it knows how to build the rest uri.